californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java

```diff
@@ -216,8 +229,18 @@ public CoapEndpoint(final InetSocketAddress address, final NetworkConfig config)
 	public CoapEndpoint(Connector connector, NetworkConfig config) {
 		this.config = config;
 		this.connector = connector;
-		this.matcher = new Matcher(config);		
-		this.coapstack = new CoapStack(config, new OutboxImpl());
+
+		if (connector.isTcp()) {
+			this.matcher = new TcpMatcher(config);
+			this.coapstack = new CoapTcpStack(config, new OutboxImpl());
+			this.serializer = new Serializer(new TcpDataSerializer());
+			this.parser = new TcpDataParser();
+		} else {
+			this.matcher = new TcpMatcher(config);
```

Comment:
Thanks Kai. I think you caught it - the tests likely failed because I fat
fingered "TcpMather" in CoapEndpoint.

I'll update and rerun.

On Tue, Jun 28, 2016 at 6:42 AM Kai Hudalla <notifications@github.com>
wrote:

> In
> californium-core/src/main/java/org/eclipse/californium/core/network/CoapEndpoint.java
> <https://github.com/eclipse/californium/pull/61#discussion_r68758158>:
>
> > @@ -216,8 +229,18 @@ public CoapEndpoint(final InetSocketAddress address, final NetworkConfig config)
> >  	public CoapEndpoint(Connector connector, NetworkConfig config) {
> >  		this.config = config;
> >  		this.connector = connector;
> > -		this.matcher = new Matcher(config);		
> > -		this.coapstack = new CoapStack(config, new OutboxImpl());
> > +
> > +		if (connector.isTcp()) {
> > +			this.matcher = new TcpMatcher(config);
> > +			this.coapstack = new CoapTcpStack(config, new OutboxImpl());
> > +			this.serializer = new Serializer(new TcpDataSerializer());
> > +			this.parser = new TcpDataParser();
> > +		} else {
> > +			this.matcher = new TcpMatcher(config);
>
> should be new UdpMatcher(config) ...
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/californium/pull/61/files/6554875f60f6f9b9c6b35ed2a14fc1f1a6c1d5bf#r68758158>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/APu4Mh6wISI3csxXK8YvTsYRpNWS8QbLks5qQSTJgaJpZM4I9Q-g>
> .
>